### PR TITLE
Propagate photon series type to imaging interfaces

### PR DESCRIPTION
**Pull request by alessandratrapani**
Following #847
Add photon_series_type input arg to imaging interfaces that do not set automatically metadata specific to TwoPhotonSeries (such as ScanImage interfaces) or specific to OnePhotonSeries (as Miniscope interface).

### CODE REVIEW

File: src/neuroconv/datainterfaces/ophys/baseimagingextractorinterface.py [hunk 1]
```diff
@@ -44,7 +45,16 @@ def __init__(
         self.verbose = verbose
         self.photon_series_type = photon_series_type
 
-    def get_metadata_schema(self) -> dict:
+    def get_metadata_schema(self, photon_series_type: Literal["OnePhotonSeries", "TwoPhotonSeries"] = None) -> dict:
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ophys/baseimagingextractorinterface.py:48]:
```suggestion
    def get_metadata_schema(self, photon_series_type: Optional[Literal["OnePhotonSeries", "TwoPhotonSeries"]] = None) -> dict:
```

I know it's just temporary; still including

File: src/neuroconv/datainterfaces/ophys/baseimagingextractorinterface.py [hunk 2]
```diff
@@ -87,7 +97,17 @@ def get_metadata_schema(self) -> dict:
         fill_defaults(metadata_schema, self.get_metadata())
         return metadata_schema
 
-    def get_metadata(self) -> DeepDict:
+    def get_metadata(self, photon_series_type: Literal["OnePhotonSeries", "TwoPhotonSeries"] = None) -> DeepDict:
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ophys/baseimagingextractorinterface.py:100]:
```suggestion
    def get_metadata(self, photon_series_type: Optional[Literal["OnePhotonSeries", "TwoPhotonSeries"]] = None) -> DeepDict:
```

**Comment by CodyCBakerPhD** [comment]
@alessandratrapani Could you add at least one test or otherwise modify the imaging extractor mixin to use this new structure so the CI doesn't get the deprecation warnings?

Should be the last thing to do on this PR then it's good to go

**Comment by alessandratrapani** [comment]
> @alessandratrapani Could you add at least one test or otherwise modify the imaging extractor mixin to use this new structure so the CI doesn't get the deprecation warnings?
> 
> Should be the last thing to do on this PR then it's good to go

I add a test to trigger the warning. I don't think the CI gets the deprecation warning because it does not pass the photn_series_type argument, but maybe I am wrong

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**